tests/rptest/services/failure_injector.py

```diff
@@ -42,16 +42,20 @@ def __exit__(self, type, value, traceback):
 
     def inject_failure(self, spec):
         self.redpanda.logger.info(f"injecting failure: {spec}")
-        self._start_func(spec.type)(spec.node)
-        if spec.length is not None:
-            if spec.length == 0:
-                self._stop_func(spec.type)(spec.node)
-            else:
-                stop_timer = threading.Timer(function=self._stop_func(
-                    spec.type),
-                                             args=[spec.node],
-                                             interval=spec.length)
-                stop_timer.start()
+        try:
+            self._start_func(spec.type)(spec.node)
+        except Exception as e:
+            self.redpanda.logger.info(f"injecting failure error: {e}")
```

Comment:
Apparently logger.exception does some unique stuff for logging details about an exception like backtrace if it is useful.